interfaces/builtin/desktop.go

```diff
@@ -609,6 +724,13 @@ func (iface *desktopInterface) AppArmorPermanentSlot(spec *apparmor.Specificatio
 	return nil
 }
 
+func (iface *desktopInterface) AppArmorConnectedSlot(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	old := "###PLUG_SECURITY_TAGS###"
+	new := plug.LabelExpression()
+	spec.AddSnippet(strings.Replace(desktopConnectedSlotAppArmor, old, new, -1))
```

Comment:
Please correct me if I am wrong, I assume this needs to be triggered only when a snap is trying to connect to relevant KDE snap that provides the desktop slot which is different from the system desktop slot?

Maybe we only want to add the relevant slot-side snippet when connecting to an app slot?
```suggestion
	// Only apply slot snippet when running as application snap
	// on classic, slot side can be system or application
	if !implicitSystemPermanentSlot(slot) {
		old := "###PLUG_SECURITY_TAGS###"
		new := plug.LabelExpression()
		spec.AddSnippet(strings.Replace(desktopConnectedSlotAppArmor, old, new, -1))
```

Reply:
done